util/findLostPatches.py

```diff
@@ -0,0 +1,176 @@
+    'description' : 'For JBIDE ' + jbide_fixversion + ': ' + taskdescriptionfull + '\n\n[Search for all task JIRA|' + tasksearch + ']\n\nSee also: ' + rootJBDS.key,
+    'issuetype' : { 'name' : 'Task' },
+    'priority' : { 'name' :'Blocker'},
+    'fixVersions' : [{ "name" : jbide_fixversion }],
+    'components' : [{ "name" : "build" }],
+    'labels' : [ "task" ]
+    }
+rootJBIDE = jira.create_issue(fields=rootJBIDE_dict)
+print("JBoss Tools       : " + jiraserver + '/browse/' + rootJBIDE.key)
+
+## map from descriptive name to list of JBIDE and/or JBDS components.
+JBT_components = {
+
+# active projects
+    "Aerogear          ": { "aerogear-hybrid", "cordovasim" },
+    "Base              ": { "common/jst/core", "usage" },
+    "Forge             ": { "forge" },
+    "Server            ": { "server" },
+    "Webservices       ": { "webservices" },
+    "Hibernate         ": { "hibernate"}, 
+    "VPE               ": { "visual-page-editor-core" },
+    "BrowserSim        ": { "browsersim" },
+    "JST               ": { "common/jst/core" },
+    "JavaEE            ": { "jsf", "seam2", "cdi", "cdi-extensions" },
+    "Central           ": { "central", "maven", "project-examples" },
+    "Arquillian        ": { "arquillian" }, # Note: s/testing-tools/arquillian/
+    "LiveReload        ": { "livereload" },
+    "OpenShift         ": { "openshift", "cdk" },
+    "Freemarker        ": { "freemarker" },
+    "Playground        ": { "playground" }, # nothing in playground currently included in JBT update site ?
+
+# disabled and probably never coming back / EOL
+    # "Birt              ": { "birt" } ,
+    #"GWT               ": { "gwt"},
+    #"Portlet           ": { "portal-gatein"},
+    #"XULRunner         ": { "xulrunner" },
+    "Discovery ": { "discovery" },
+    }
+
+# Currently, the repo url (for printing links to the missing commits)
+# is calculated as  "https://github.com/jbosstools/jbosstools-$key.git"
+# If this is wrong, it will need to be refactored.
+
+def nametuple(x):
+    return { "name" : x }
+
+def quote(x):
+    return '"' + x + '"'
+
+
+# We've made the first parent jira. 
+# To avoid cloning all repos here in this folder,
+# lets make a sub-directory to do our work in
+workingdir = os.getcwd();
+tmpsubdir = workingdir + "/tmplostpatches"
+try:
+   os.stat(tmpsubdir)
+except:
+   os.mkdir(tmpsubdir)
+
+os.chdir(tmpsubdir)
+
+
+for name, comps in JBT_components.iteritems():
+    
+    cms = map(nametuple, comps)    
+    #print name + "->" + str(cms)
+    workingdir = os.getcwd()
+    subfoldername="jbosstools-" + name.lower()
+    githubrepo = "https://github.com/jbosstools/" + subfoldername.strip() + ".git"
+    print githubrepo;
+    os.system("git clone " + githubrepo)
+    os.chdir(workingdir + "/" + subfoldername.strip())
+    #  TODO externalize the branch here
+    p = Popen(['bash', '../../findLostPatchesOneRepository.sh', 'jbosstools-4.3.x'], stdout=PIPE, stderr=PIPE, stdin=PIPE)
```

Comment:
This is the last piece that I believe needs 100% to be externalized, but I'm not quite sure where / how to put it.  @nickboldt  can you help here?